src/index.ts

```diff
@@ -1,6 +1,41 @@
-function atrapaAlRaton(text: string): boolean {
-    if (text.length > 5) {
-        return true;
-    } else return false
-}
-export default atrapaAlRaton
\ No newline at end of file
+export const atrapaAlRaton = (patron: string): boolean => {
+    if (!correctPatron(patron)) {
+        throw new Error('Patrón Incorrecto');
+    }
+
+    return patron.length > 5;
+};
+
+export const correctPatron = (patron: string): boolean => {
+    if (typeof patron !== 'string') {
+        return false;
+    }
+
+    if (patron.length < 3) {
+        return false;
+    }
+
+    if (patron[0] !== 'C' || patron[patron.length - 1] !== 'M') {
+        return false;
+    }
+
+    const caracteresMedio = patron.slice(1, -1);
+    for (const char of caracteresMedio) {
```

Comment:
Buenísimo, me gusta esta solución 😃

Reply:
Sólo como comentario, si quisieras usar una aproximación más funcional, hay algún método de array que puedas usar para no usar el for? pensá que caracteresMedio es un array.

Reply:
En ese caso utilizaria filtrer

Reply:
Perdón, slice devuelve otro string, no un array. Dejalo así como está, está bien.